Docs for running docker-compose locally with SDK and server (#3390)
googleforgames/agones: site/content/en/docs/Guides/Client SDKs/local.md

docker run --network=host --rm -v $(pwd)/gameserver.yaml:/tmp/gameserver.yaml us-docker.pkg.dev/agones-images/release/agones-sdk:{{<release-version>}} --local -f /tmp/gameserver.yaml
```

> If you would like to run the ClientSDK and your game server container together with Docker Compose create a `docker-compose.yaml` file.

Reviewer: Do we want to add an extra note here that's something like...

```suggestion
if you run Docker on a OS that doesn't run Docker natively or in a VM, such as on Windows or macOS., you may want to to run the ClientSDK and your game server container together with Docker Compose . To do so, create a `docker-compose.yaml` file setup with a network overlay shared between them:
```

Just so people have a _why_ on they might want to do this.

WDYT?

Author: I like, makes sense. Updated in latest commit.

```yaml
version: '3'
services:
  gameserver:

>     build: . # <location to your image here>

Reviewer: Is `build` a path to the image, or a path to the build context? 

(Just looking at https://docs.docker.com/compose/compose-file/compose-file-v3/#build ?)

Author: Ah, good question. `build` is to the context, but also defaults to assuming `Dockerfile` is at the root of that path. Happy to split this out to make more explicit for the docs like what is shown in that second example in your link. What do you think?

Author: Updated verbiage here to `# <path to build context>`

> If you run Docker on a OS that doesn't run Docker natively or in a VM, such as on Windows or macOS, you may want to to run the ClientSDK and your game server container together with Docker Compose. To do so, create a `docker-compose.yaml` file setup with a network overlay shared between them:

Reviewer: ```suggestion
If you run Docker on a OS that doesn't run Docker natively or in a VM, such as on Windows or macOS, you may want to to run the ClientSDK and your game server container together with [Docker Compose](https://docs.docker.com/compose/). To do so, create a `docker-compose.yaml` file setup with a network overlay shared between them:
```

One final improvement I'd like to see - otherwise, this is good to go 👍🏻

    ports:
      - "127.0.0.1:7777:7777/udp"

  sdk-server:

>     image: "us-docker.pkg.dev/agones-images/release/agones-sdk:1.34.0"

Reviewer: ```suggestion
    image: "us-docker.pkg.dev/agones-images/release/agones-sdk:{{<release-version>}}"
```

Should work (I think 😃)

Author: fixed

    command: --local -f /gs_config

>     network_mode: service:gameserver

Reviewer: Wondering if we want an extra comment here explaining that this is the shared network?

Author: It's a good idea. I added a comment in the code block pointing this out: `# <shared network between sdk and game server>`
